app/src/main/java/io/gnosis/safe/di/modules/RepositoryModule.kt

```diff
@@ -51,20 +53,86 @@ class RepositoryModule {
     @Provides
     @Singleton
     fun providesDomainResolutionLibrary(): DomainResolution {
-        val firstWord = BuildConfig.BLOCKCHAIN_NET_URL.removePrefix("https://").split(".").first();
-        val network = Network.valueOf(firstWord.toUpperCase());
-        return Resolution.builder()
+        val firstWord = BuildConfig.BLOCKCHAIN_NET_URL.removePrefix("https://").split(".").first()
+        val network = Network.valueOf(firstWord.toUpperCase())
+        return try {
+            Resolution.builder()
                 .chainId(NamingServiceType.CNS, network)
                 .infura(NamingServiceType.CNS, BuildConfig.INFURA_API_KEY)
                 .build()
+        } catch (throwable: Throwable) {
```

Comment:
Is there a better way than providing an "empty" interface? If not, then I'd move this to separate file called DummyDomainResolution.